sdk/src/asset_handlers/sfnt_io.rs

```diff
@@ -1300,17 +1258,16 @@ where
 /// A result containing the `C2PA` font table data
 fn read_c2pa_from_stream<T: Read + Seek + ?Sized>(reader: &mut T) -> Result<TableC2PA> {
     let sfnt = SfntFont::from_reader(reader).map_err(|_| Error::FontLoadError)?;
-    let c2pa_table: Option<TableC2PA> = match sfnt.tables.get(&C2PA_TABLE_TAG) {
-        None => None,
-        Some(ostensible_c2pa_table) => match ostensible_c2pa_table {
-            Table::C2PA(bonafied_c2pa_table) => Some(bonafied_c2pa_table.clone()),
-            _ => {
-                // Non-C2PA table with C2PA tag
-                return Err(Error::FontLoadError);
-            }
-        },
-    };
-    c2pa_table.ok_or(Error::JumbfNotFound)
+    match sfnt.tables.get(&C2PA_TABLE_TAG) {
+        None => Err(Error::JumbfNotFound),
+        // If there is, replace its `manifest_store` value with the
+        // provided one.
+        Some(Table::C2PA(c2pa)) => Ok(c2pa.clone()),
+        // Yikes! Non-C2PA table with C2PA tag!
+        Some(_) => {
+            return Err(Error::FontLoadError);
```

Comment:
So we actually don't need `return` here, because `Err()` automatically returns the error type (clippy is upset!)

Reply:
(huh, how did I push if clippy hates it... toolchain?)